src/scripts/exclusion_term_expansion/exclusion_term_expansion.py

```diff
@@ -0,0 +1,299 @@
+    export_fields=['term_id', 'exclusion_reason'], uri_fields=['term_id', 'child_id']
+) -> pd.DataFrame:
+    """Using an ontology file and an exclusions file, query and get return an extensional list of exclusions.
+    cache_suffix: Used by and explained in sparql_jinja2_robot_query()."""
+    # Variable names: kids0=Children excluded; kids1=Children included
+    df_kids1 = exclusions_df[exclusions_df['exclude_children'] == False]
+    df_kids0 = exclusions_df[exclusions_df['exclude_children'] == True]
+    terms_kids1: List[str] = list(set(df_kids1['term_id']))  # QA: set() removes any duplicates
+    df_kids1_results: pd.DataFrame = sparql_jinja2_robot_query(
+        terms=terms_kids1,
+        query_template_path=SPARQL_CHILD_INCLUSION_PATH,
+        onto_path=onto_path,
+        cache_suffix=cache_suffix)
+
+    # Massage results
+    # # Convert URI back to prefix
+    for fld in uri_fields:
+        df_kids1_results[fld] = df_kids1_results[fld].apply(uri_to_curie)
+    # # JOIN: To get exclusion_reason
+    df_kids1_results2 = pd.merge(df_kids1_results, exclusions_df, how='left', on='term_id').fillna('')
+    # # Drop parent term data
+    df_kids1_results2 = df_kids1_results2[['child_id', 'child_label', 'exclusion_reason']]
+    # # Rename child fields as they are now top-level terms
+    df_kids1_results2 = df_kids1_results2.rename(columns={
+        'child_id': 'term_id',
+        'child_label': 'term_label',
+    })
+    # # Drop unneeded fields
+    df_kids0 = df_kids0[export_fields]
+    df_kids1_results2 = df_kids1_results2[export_fields]
+
+    # CONCAT: Combine term child results w/ terms where children were excluded
+    df_results = pd.concat([df_kids0, df_kids1_results2]).fillna('')
+
+    return df_results
+
+
+def run(onto_path: str, exclusions_path: str, save=CONFIG['save']) -> Dict[str, pd.DataFrame]:
+    """Run"""
+    # Vars
+    df_explicit_results = None
+    df_regex_results = None
+    exclusion_fields = ['exclusion_reason', 'exclude_children']
+    sep = '\t' if exclusions_path.endswith('.tsv') else ',' if exclusions_path.endswith('.csv') else ''
+    onto_name = os.path.basename(onto_path).split('.')[0]
+    df_results_path = os.path.join(CONFIG_DIR, f'{onto_name}_exclusion_reasons.tsv')
+    df_results_simple_path = os.path.join(CONFIG_DIR, f'{onto_name}_term_exclusions.txt')
+
+    # Prepare data
+    df = pd.read_csv(exclusions_path, sep=sep).fillna('')
+    # - Split into two: 1 with regexp patterns, and 1 with explicit term_id
+    df_regex = df[df['term_label'].str.startswith(CONFIG['regex_prefix'])]
+    df_explicit = df.filter(items=set(df.index) - set(df_regex.index), axis=0)
+
+    # Query 1: on explicit term_id
+    if len(df_explicit) > 0:
+        df_explicit_results = expand_ontological_exclusions(onto_path, df_explicit, cache_suffix='1')
+
+    # Query 2: on regex pattern
+    # todo: later: #x4: use OAK when `.ttl` is supported: https://github.com/INCATools/ontology-access-kit/issues/134
```

Comment:
This block of code looks a little messy, but it does the job. I use the Jinja template to do 1 query per REGEX pattern, then combine the results.

I tried to use OAK, but `.ttl` doesn't seem to be supported yet, so I made a ticket: https://github.com/INCATools/ontology-access-kit/issues/134

Reply:
It is, just not with pronto. Responded on issue for recommended procedure.